src/build/__init__.py

```diff
@@ -329,6 +338,36 @@ def build(self, distribution, output_directory, config_settings=None, metadata_d
         kwargs = {} if metadata_directory is None else {'metadata_directory': metadata_directory}
         return self._call_backend(build, output_directory, config_settings, **kwargs)
 
+    def metadata_path(self, output_directory):  # type: (str) -> str
+        """
+        Generates the metadata directory of a distribution and returns its path.
+
+        If the backend does not support the ``prepare_metadata_for_build_wheel``
+        hook, a wheel will be built and the metadata extracted.
+
+        :param output_directory: Directory to put the metadata distribution in
+        """
+        # prepare_metadata hook
+        metadata = self.prepare('wheel', output_directory)
+        if metadata:
```

Comment:
Nitpicking but if `metadata is not None`.  In the unlikely case that the backend returns an empty string we'd be masking their error.